### PR TITLE
Bump Psalm level to 3

### PR DESCRIPTION
**Pull request by morozov**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no

### CODE REVIEW

File: lib/Doctrine/DBAL/Driver/IBMDB2/Exception/PrepareFailed.php [hunk 1]
```diff
@@ -13,8 +13,17 @@
...
+    /**
+     * @psalm-param array{message: string}|null $error
+     */
+    public static function new(?array $error): self
     {
+        if ($error !== null) {
+            $message = $error['message'];
+        } else {
+            $message = 'Unknown error';
+        }
```
Comment (greg0ire) [inline lib/Doctrine/DBAL/Driver/IBMDB2/Exception/PrepareFailed.php:25]: Let's avoid the negative logic and the `else`:
```suggestion
        if ($error === null) {
            return new self('Unknown error');
        }
           
        return new self($error['message']);
```

File: tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php [hunk 2]
```diff
@@ -682,14 +680,12 @@ public function testBitComparisonExpressionSupport(): void
...
-        $sql[] = 'SELECT ';
-        $sql[] = 'test_int, ';
-        $sql[] = 'test_string, ';
-        $sql[] = $platform->getBitOrComparisonExpression('test_int', 2) . ' AS bit_or, ';
-        $sql[] = $platform->getBitAndComparisonExpression('test_int', 2) . ' AS bit_and ';
-        $sql[] = 'FROM fetch_table';
+        $sql = 'SELECT test_int, test_string'
+            . ', ' . $platform->getBitOrComparisonExpression('test_int', 2) . ' AS bit_or'
+            . ', ' . $platform->getBitAndComparisonExpression('test_int', 2) . ' AS bit_and'
+            . ' FROM fetch_table';
```
Comment (greg0ire) [inline tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php:693]:
```suggestion
        $sql = sprintf(
        <<<'SQL'
        SELECT
            test_int,
            test_string,
            %s AS bit_or,
           %s as bit_and
        FROM fetch_table
SQL
,
            $platform->getBitOrComparisonExpression('test_int', 2), 
            $platform->getBitAndComparisonExpression('test_int', 2)
        );
```
modulo the cs issues of course

**Review by greg0ire — COMMENTED** [review]
Please check Slack, I'd like to talk about merges up.

File: tests/Doctrine/Tests/DBAL/ConnectionTest.php [hunk 3]
```diff
@@ -51,7 +51,7 @@ protected function setUp(): void
     }
 
     /**
-     * @return Connection|MockObject
+     * @return Connection&MockObject
```
[the remark this answers is not in the document]
Comment (morozov) [reply tests/Doctrine/Tests/DBAL/ConnectionTest.php:54]: This doesn't seem to work with the latest version of dependencies:
```
$ composer show | grep psalm

psalm/plugin-phpunit                           0.10.1  Psalm plugin for PHPUnit
vimeo/psalm                                    3.17.2  A static analysis tool for finding...

$ psalm

...
ERROR: InvalidArgument - tests/Doctrine/Tests/DBAL/Types/VarDateTimeTest.php:67:71 - Argument 2 of Doctrine\DBAL\Types\VarDateTimeType::convertToPHPValue expects Doctrine\DBAL\Platforms\AbstractPlatform, PHPUnit\Framework\MockObject\MockObject<Doctrine\DBAL\Platforms\AbstractPlatform> provided (see https://psalm.dev/004)
        self::assertSame($date, $this->type->convertToPHPValue($date, $this->platform));


------------------------------
251 errors found
------------------------------
```

**Review by greg0ire — APPROVED** [review]

**Pull request merged**